fastlane/Fastfile

```diff
@@ -353,51 +309,27 @@ platform :ios do
       new_version: release_version_current,
       release_notes_file_path: RELEASE_NOTES_SOURCE_PATH
     )
-  end
-
-  # Executes the final steps for the code freeze
-  #
-  #  - Generates `.strings` files from code then merges the other, manually-maintained `.strings` files with it
-  #  - Triggers the build of the first beta on CI
-  #
-  # @option [Boolean] skip_confirm (default: false) If true, avoids any interactive prompt
-  #
-  desc 'Completes the final steps for the code freeze'
-  desc 'Creates a new release branch from the current trunk'
-  lane :complete_code_freeze do |options|
-    require_env_vars!('GITHUB_TOKEN')
-
-    # Verify that the current branch is a release branch. Notice that `ensure_git_branch` expects a RegEx parameter
-    ensure_git_branch(branch: '^release/')
-
-    # Verify that there's nothing in progress in the working copy
-    ensure_git_status_clean
-
-    UI.important("Completing code freeze for: #{release_version_current}")
-    UI.user_error!('Aborted by user request') unless options[:skip_confirm] || UI.confirm('Do you want to continue?')
 
     generate_strings_file_for_glotpress
-    create_merge_release_branch
 
-    after_confirming_push(push_merge_branch: true) do
+    after_confirming_push do
       trigger_beta_build(branch_to_build: release_branch_name)
+      create_backmerge_pr
```

Comment:
Before this change, the `create_merge_release_branch` call only created the intermediate `merge/…` branch—and pushed it to remote during `after_confirming_push(push_merge_branch: true)`—but didn't open the backmerge PR at the time of the call to this `code_freeze` lane yet.

Instead, after the intermediate branch was created and pushed to the remote, the beta build was triggered, which ultimately called the `build_and_upload_app_store_connect` lane on CI, and that's [only in that lane that the backmerge PR was created](https://github.com/Automattic/pocket-casts-ios/pull/2215/files#diff-dff4b99d4e651ab9d7597876ec8dbe92aa934e42c0fb55f4aadd6c106fc96688L530-L541)—only after the CI had finished building the `.ipa` and uploaded it to TestFlight!

This PR changes the approach to not have the `build_and_upload_app_store_connect` lane be the one creating the backmerge PR, but instead have the lanes that are triggering those beta/release builds (namely `code_freeze` and `new_beta_release` which are calling `trigger_beta_build`, and `finalize_release` and `finalize_hotfix_release` which are calling `trigger_release_build`) be the ones also creating the PR at the same time.
This not only will allow the Release Manager to review the backmerge PR (and solve any potential conflict on it) sooner—without having to wait multiple minutes for the beta/release build to have finished building on CI for the PR to be created—but will also mean that the CI checks on that backmerge PR (unit tests, etc) will start sooner and run in parallel to the beta/release CI build, thus having that backmerge PR go green and ready to be merged sooner too.

Reply:
For a future PR I guess: do we even need `after_confirming_push`?

Reply:
Nope, I'm planning to remove this as well, but in a future PR indeed, as I didn't want to change too much at a time (it already gave me such headaches to wrap my head around those various helpers and custom implementations used in PC's Fastfile).

I'm probably gonna remove this when I'll get closer to finally have those lanes be run by CI, at which point all the confirmation points will be removed (or at least use `skip_confirm: true`). Even if we keep the possibility for confirmation (in case we want to still sometimes call those lanes from our Macs for debugging purposes) I'm planning to get rid of that `after_confirming_push` method and clean that up at some point anyway indeed to make the confirmation points, if we keep them, more consistent across the `Fastfile` and with our other repos.